apiserver/pkg/chat/chat_server.go

```diff
@@ -206,4 +204,184 @@ func (cs *ChatServer) GetMessageReferences(ctx context.Context, req MessageReqBo
+		if err != nil {
+			return nil, err
+		}
+	Outer:
+		for _, detail := range kb.Status.FileGroupDetail {
+			if detail.Source == nil || detail.Source.Name == "" {
+				continue
+			}
+			obj, err := c.Resource(schema.GroupVersionResource{Group: v1alpha1.GroupVersion.Group, Version: v1alpha1.GroupVersion.Version, Resource: "versioneddatasets"}).
+				Namespace(detail.Source.GetNamespace(kb.Namespace)).Get(ctx, detail.Source.Name, metav1.GetOptions{})
+			if err != nil {
+				klog.Infof("failed to get versionedDataset: %s, try next one", err)
+				continue
+			}
+			versionedDataset := &v1alpha1.VersionedDataset{}
+			err = runtime.DefaultUnstructuredConverter.FromUnstructured(obj.UnstructuredContent(), versionedDataset)
+			if err != nil {
+				klog.Infof("failed to convert versionedDataset: %s, try next one", err)
+				continue
+			}
+			if !versionedDataset.Status.IsReady() {
+				klog.Infof("versionedDataset is not ready, try next one")
+				continue
+			}
+			info := &v1alpha1.OSS{Bucket: versionedDataset.Namespace}
+			for _, fileDetails := range detail.FileDetails {
+				info.Object = filepath.Join("dataset", versionedDataset.Name, versionedDataset.Spec.Version, fileDetails.Path)
+				file, err := ds.ReadFile(ctx, info)
+				if err != nil {
+					klog.Infof("failed to read file: %s, try next one", err)
+					continue
+				}
+				defer file.Close()
+				data, err := io.ReadAll(file)
+				if err != nil {
+					klog.Infof("failed to read file: %s, try next one", err)
+					continue
+				}
+				dataReader := bytes.NewReader(data)
+				doc, err := pkgdocumentloaders.NewQACSV(dataReader, "").Load(ctx)
+				if err != nil {
+					klog.Infof("failed to load doc file: %s, try next one", err)
+					continue
+				}
+				for i := 0; i < remains && i < len(doc); i++ {
+					promptStarters = append(promptStarters, doc[i].PageContent)
+				}
+				remains = limit - len(promptStarters)
+				if remains == 0 {
+					break Outer
+				}
+			}
+		}
+	} else {
+		klog.V(3).Infoln("app has no knowlegebase, let llm generate some question")
+		if model != nil {
+			p := prompts.NewPromptTemplate(PromptForGeneratePromptStarters, []string{"limit", "displayName", "description"})
+			var c *chains.LLMChain
+			if len(chainOptions) > 0 {
+				c = chains.NewLLMChain(model, p, chainOptions...)
+			} else {
+				c = chains.NewLLMChain(model, p)
+			}
+			result, err := chains.Predict(ctx, c,
+				map[string]any{
+					"limit":       limit,
+					"displayName": app.Spec.DisplayName,
+					"description": app.Spec.Description,
+				},
+			)
+			if err != nil {
+				return nil, err
+			}
+			res := strings.Split(result, "\n")
+			for _, r := range res {
+				promptStarters = append(promptStarters, strings.TrimSpace(r))
+			}
+		}
+	}
+	return promptStarters, nil
+}
+
+const PromptForGeneratePromptStarters = `You are the friendly and curious questioner, please ask {{.limit}} questions based on the name and description of this app below.
```

Comment:
any idea about this prompt?

Reply:
looks good enough, here is what AI suggests.
```
You are the friendly and curious questioner, your task is to ask 3 thoughtful questions based on the name and description of this app.

Name of the application: 英语老师
Description: 我将跟你交流英语学习中的各类问题，包括词汇、语法、文化等。

Requires language consistent with the name and description of the application, no restating of my words, questions only, one question per line, no subheadings.

The question you asked is:
```